src/app/ngrx/effects/castle.effects.ts

```diff
@@ -287,24 +281,24 @@ export class CastleEffects {
     ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_RANKS_LOAD),
     map((action: ACTIONS.CastlePodcastRanksLoadAction) => action.payload),
     mergeMap((payload: ACTIONS.CastlePodcastRanksLoadPayload) => {
-      const { id, interval, group, filter, beginDate, endDate } = payload;
+      const { podcastId, interval, group, filter: groupFilter, beginDate, endDate } = payload;
```

Comment:
I love how destructuring renaming makes a rename for clarity like this so easy.

Reply:
`groupFilter` is def more clear naming, but I was honestly doing it because it came up in linting. I was shadowing RxJS filter. Oops! If I had it to do over again, the `RouterParams` `filter` would be called `groupFilter`. You only get so many do overs per PR.